src/main/java/teammates/common/datatransfer/InstructorAttributes.java

```diff
@@ -324,18 +314,9 @@ public boolean hasTutorPrivileges() {
         return privileges.hasTutorPrivileges();
     }
 
-    /**
-     * pre-condition: instructorPrivilegesAsText and privileges should be non-null
-     * @param instructor
-     * @return
-     */
     public boolean isEqualToAnotherInstructor(InstructorAttributes instructor) {
-        if (gson.toJson(this).equals(gson.toJson(instructor))) {
-            return true;
-        } else {
-            return !instructorPrivilegesAsText.equals(instructor.instructorPrivilegesAsText) 
-                   && privileges.equals(instructor.privileges);
-        }
+        JsonParser parser = new JsonParser();
+        return parser.parse(getJsonString()).equals(parser.parse(instructor.getJsonString()));
```

Comment:
If `jsonString` is converted from an `InstructorAttributes` object, calling `fromJson(jsonString, JsonElement.class)` will convert `jsonString` into a `JsonElement` object. Since `JsonElement` is not the same as `InstructorAttributes`, gson cannot parse `jsonString` correctly.